pkgs/applications/misc/navit/default.nix

```diff
@@ -68,10 +68,10 @@ stdenv.mkDerivation rec {
   '';
 
   # TODO: fix upstream?
+  libPath = stdenv.lib.makeLibraryPath ([ stdenv.cc.libc ] ++ buildInputs );
```

Comment:
I prefer not to set this kind of thing as attributes of the derivation but instead do something like:
```nix
postFixup = let
  libPath = stdenv.lib.makeLibraryPath ([ stdenv.cc.libc ] ++ buildInputs );
''
find "$out/lib" -type f -name "*.so" -exec patchelf --set-rpath ${libPath} {} \;
'';
```

It makes for more readable instantiated derivations.

Reply:
which means, that nix variables do not become environment variables in the build.

Reply:
Oki, seems good on paper, but just doesn't work.

genesis@genlaptop:/nix/store/y27wr1g1hw6jxbrc13s1jxh52a9mk6xw-navit-0.5.2/lib]$ for f in $(find -iname "*.so"); do echo $f &&  ldd -v $f | grep not  ; done
./navit/font/libfont_freetype.so
./navit/graphics/libgraphics_gtk_drawing_area.so
        libcairo.so.2 => not found
./navit/graphics/libgraphics_opengl.so
./navit/graphics/libgraphics_sdl.so
./navit/graphics/libgraphics_null.so
./navit/gui/libgui_gtk.so
./navit/gui/libgui_internal.so
./navit/binding/libbinding_dbus.so
        libglib-2.0.so.0 => not found
./navit/binding/libbinding_python.so
./navit/speech/libspeech_dbus.so
        libglib-2.0.so.0 => not found
./navit/speech/libspeech_cmdline.so
./navit/vehicle/libvehicle_gpsd_dbus.so
        libglib-2.0.so.0 => not found
./navit/vehicle/libvehicle_demo.so
./navit/vehicle/libvehicle_file.so
./navit/vehicle/libvehicle_pipe.so
./navit/vehicle/libvehicle_serial.so
./navit/vehicle/libvehicle_socket.so
./navit/osd/libosd_core.so
./navit/map/libmap_binfile.so
./navit/map/libmap_filter.so
./navit/map/libmap_mg.so
./navit/map/libmap_shapefile.so
./navit/map/libmap_textfile.so
./navit/map/libmap_csv.so